File: java-playground/src/test/java/org/gonza/javaplayground/game/car/CarNameTest.java
package org.gonza.javaplayground.game.car;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;

public class CarNameTest {
    @Test
    public void 이름이_5보다_길면_예외가_발생한다() {
        String longName = "too-long-name";
        assertThrows(IllegalArgumentException.class, () -> new CarName(longName));
    }

    @Test
    public void 이름의_실제_값을_알_수_있다() {
        String name = "test1";
        CarName carName = new CarName(name);
        assertEquals(name, carName.getValue());
    }
}

File: kotlin-playground/src/test/kotlin/org/gonza/kotlinplayground/CarNameTest.kt
package org.gonza.kotlinplayground

import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test

class CarNameTest {
    @Test
    fun `이름은 5자를 초과할 수 없다`() {
        Assertions
            .assertThatThrownBy { CarName("123456") }
            .isInstanceOf(IllegalArgumentException::class.java)
            .hasMessage("이름은 5자를 초과할 수 없습니다.")
    }

    @Test
    fun `이름의 길이는 양쪽 빈값을 제외한 길이가 5를 초과하지 않는다`() {
        val carName = CarName(" 1234 ")
        val expectedValue = CarName(" 1234 ")

        Assertions.assertThat(carName).isEqualTo(expectedValue)
    }
}
